ts/src/bingx.ts

```diff
@@ -0,0 +1,2342 @@
+         * @param {float} amount the amount to set the margin to
+         * @param {object} params parameters specific to the bingx api endpoint
+         * @returns {object} A [margin structure]{@link https://docs.ccxt.com/#/?id=add-margin-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const type = this.safeInteger (params, 'type'); //  1 increase margin 2 decrease margin
+        this.checkRequiredArgument ('setMargin', type, 'type');
+        const request = {
+            'symbol': market['id'],
+            'amount': this.amountToPrecision (market['symbol'], amount),
+            'type': type,
+        };
+        const response = await this.swapV2PrivatePostTradePositionMargin (this.extend (request, params));
+        //
+        //    {
+        //        "code": 0,
+        //        "msg": "",
+        //        "amount": 1,
+        //        "type": 1
+        //    }
+        //
+        return response;
+    }
+
+    async fetchLeverage (symbol: string, params = {}) {
+        /**
+         * @method
+         * @name bingx#fetchLeverage
+         * @description fetch the set leverage for a market
+         * @see https://bingx-api.github.io/docs/#/swapV2/trade-api.html#Query%20Leverage
+         * @param {string} symbol unified market symbol
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @returns {object} a [leverage structure]{@link https://docs.ccxt.com/#/?id=leverage-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        const response = await this.swapV2PrivateGetTradeLeverage (this.extend (request, params));
+        //
+        //    {
+        //        "code": 0,
+        //        "msg": "",
+        //        "data": {
+        //            "longLeverage": 6,
+        //            "shortLeverage": 6
+        //        }
+        //    }
+        //
+        return response;
+    }
+
+    async setLeverage (leverage, symbol: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name bingx#setLeverage
+         * @description set the level of leverage for a market
+         * @see https://bingx-api.github.io/docs/swapV2/trade-api.html#_12-switch-leverage
+         * @param {float} leverage the rate of leverage
+         * @param {string} symbol unified market symbol
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @returns {object} response from the exchange
+         */
+        this.checkRequiredSymbol ('setLeverage', symbol);
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const side = this.safeStringUpper (params, 'side');
+        this.checkRequiredArgument ('setLeverage', side, 'side');
```

Comment:
first we should `checkRequiredArgument` before actually getting it. Second `checkRequiredArgument` accepts a list of options to help the user, example: `this.checkRequiredArgument ('setLeverage', side, 'side', ['buy', 'sell']);`

Reply:
This way?
```js
this.checkRequiredArgument ('setLeverage', this.safeStringUpper (params, 'side'), 'side', [ 'LONG', 'SHORT' ]);
```

Or like this?
```js
const side = this.safeStringUpper (params, 'side');
this.checkRequiredArgument ('setLeverage', side, 'side', [ 'LONG', 'SHORT' ]);
```